src/pair_rann.h

```diff
@@ -0,0 +1,171 @@
+this document was conducted at CAVS, MSU.  Permission was granted by ERDC
+to publish this information. Any opinions, findings and conclusions or
+recommendations expressed in this material are those of the author(s) and
+do not necessarily reflect the views of the United States Army.​”
+
+DISTRIBUTION A. Approved for public release; distribution unlimited. OPSEC#4918
+ */
+
+#ifdef PAIR_CLASS
+
+PairStyle(rann,PairRANN)
+
+#else
+
+#ifndef LMP_PAIR_RANN
+#define LMP_PAIR_RANN
+
+#define MAXLINE 1024
+
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include "atom.h"
+#include "force.h"
+#include "comm.h"
+#include "memory.h"
+#include "neighbor.h"
+#include "neigh_list.h"
+#include "neigh_request.h"
+#include "memory.h"
+#include "error.h"
+#include "update.h"
+#include "pair.h"
+#include <map>
+#include <string>
+
+namespace LAMMPS_NS {
+
+class PairRANN : public Pair {
+ public:
+
+  //inherited functions
+  PairRANN(class LAMMPS *);
+  ~PairRANN();
+  void compute(int, int);
+  void settings(int, char **);
+  void coeff(int, char **);
+  void init_style();
+  double init_one(int, int);
+  void init_list(int , NeighList *);
+  void errorf(const char*);
+  int count_words(char *);
+  //black magic for modular fingerprints and activations
+  class Activation ***activation;
+  class Fingerprint ***fingerprints;
+  typedef Fingerprint *(*FingerprintCreator)(PairRANN *);
+  typedef Activation *(*ActivationCreator)(PairRANN *);
+  typedef std::map<std::string,FingerprintCreator> FingerprintCreatorMap;
+  typedef std::map<std::string,ActivationCreator> ActivationCreatorMap;
+  FingerprintCreatorMap *fingerprint_map;
+  ActivationCreatorMap *activation_map;
+  Fingerprint * create_fingerprint(const char *);
+  Activation * create_activation(const char *);
+
+  //global variables
+  int nelements;                // # of elements (distinct from LAMMPS atom types since multiple atom types can be mapped to one element)
+  int nelementsp;				// nelements+1
+  char **elements;              // names of elements
+  char **elementsp;				// names of elements with "all" appended as the last "element"
+  double *mass;                 // mass of each element
+  double cutmax;				// max radial distance for neighbor lists
+  int *map;                     // mapping from atom types to elements
+  int *fingerprintcount;		// static variable used in initialization
+  int *fingerprintlength;       // # of input neurons defined by fingerprints of each element.
+  int *fingerprintperelement;   // # of fingerprints for each element
+  bool doscreen;//screening is calculated if any defined fingerprint uses it
+  bool allscreen;//all fingerprints use screening so screened neighbors can be completely ignored
+  bool dospin;
+  int res;//Resolution of function tables for cubic interpolation.
+  int memguess;
+	double *screening_min;
+	double *screening_max;
+	bool **weightdefined;
+	bool **biasdefined;
+
+	struct Simulation{
```

Comment:
this and the following struct do not need to be define here. a forward declaration should be sufficient. the full struct can be defined in the implementation.

Reply:
I tried to get this to work without success. I was able to compile pair_rann.cpp with it defined as you suggest, but the fingerprint files would not compile since they also reference the structs. Maybe I'm not doing it correctly.